File: CPSC411/Objective-C/IMail/IMail/ViewController.m
//
//  ViewController.m
//  InstaEmail
//
//  Created by William McCarthy on 19/7/2022.
//

#import "ViewController.h"

@implementation ViewController

@synthesize emailPicker=emailPicker_;
@synthesize notesField=notesField_;


- (void)viewDidLoad {
  [super viewDidLoad];
  activities_ = [[NSArray alloc] initWithObjects:@"sleeping", @"eating", @"working",
                 @"thinking", @"crying", @"begging",
                 @"leaving", @"shopping", @"writing",
                 @"reading", @"coding", @"philosophizing",
                 nil];
  feelings_ = [[NSArray alloc] initWithObjects:@"awesome", @"sad", @"happy",
               @"ambivalent", @"inspired", @"creative",
               @"nauseous", @"psyched", @"confused",
               @"hopeful", @"anxious", @"logical",
               nil];
  // Do any additional setup after loading the view.
}

#pragma mark -
#pragma mark Picker Datasource Protocol

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *) pickerView {
  return 2;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
  return [(component == 0 ? activities_ : feelings_) count];
}

#pragma mark -
#pragma mark Picker Delegate Protocol

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
  return [(component == 0 ? activities_ : feelings_) objectAtIndex:row];
  return nil;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponenet:(NSInteger)component {
  return [(component == 0 ? activities_ : feelings_) count];
}

#pragma mark -
#pragma mark Actions

- (IBAction) sendButtonTapped: (id)sender {
  NSLog(@"Email button pressed!");

  NSString* theMessage = [NSString stringWithFormat:
                          @"%@ I'm %@ and feeling %@ about it. Just letting you know!",
                          notesField_.text ? notesField_.text : @"",
                          [activities_ objectAtIndex:[emailPicker_ selectedRowInComponent:0]],
                          [feelings_ objectAtIndex:[emailPicker_ selectedRowInComponent:1]]];
  NSLog(@"%@", theMessage);
  [self sendEmail:theMessage];
}

-(void) sendEmail: (NSString*)msg {
  if ([MFMailComposeViewController canSendMail]) {
    MFMailComposeViewController* ctrl = [[MFMailComposeViewController alloc] init];
    ctrl.mailComposeDelegate = self;
    [ctrl setToRecipients:@[@"willandmei@gmail.com"]];
    [ctrl setCcRecipients:nil];
    [ctrl setBccRecipients:nil];
    [ctrl setSubject:@"Hello, Mei!"];
    [ctrl setMessageBody:msg isHTML:NO];
//    [self presentModalViewController:ctrl animated:YES];
    
    [self presentModalViewController:ctrl animated:YES];
//    [ctrl release];
  } else {
    NSLog(@"%@", @"Sorry, you need to setup mail first!");
  }
}

-(IBAction) textFieldDoneEditing: (id) sender {
  [sender resignFirstResponder];
}

#pragma mark -
#pragma mark Mail composer delegate method

-(void)mailComposeController:(MFMailComposeViewController*) controller
         didFinishWithResult:(MFMailComposeResult)result error:(NSError*)error {
  [self dismissModalViewControllerAnimated:YES];
}

- (void) dealloc {
//  [emailPicker_ release];    // ARC handles this now
//  [activities_ release];
//  [feelings_ release];
//  [notesField_ release];
//  [super dealloc];
}


@end

File: CPSC411/Swift Stuff/Calculatorv2/Calculatorv2/ViewController.swift
//
//  ViewController.swift
//  Calculatorv2
//
//  Created by csuftitan on 10/18/22.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var label: UILabel!
    var numberOnScreen:Double = 0
    var previousNumber:Double = 0
    var performingMath:Bool = false
    var operation = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func numbers(_ sender: UIButton) {
        if performingMath == true {
            label.text = ""
            label.text = label.text! + String(sender.tag-1)
            performingMath = false
        }
        else {
            label.text = label.text! + String(sender.tag-1)
        }
    }
    
    @IBAction func buttons(_ sender: UIButton) {
        if sender.tag != 11 && sender.tag != 16 {
            previousNumber = Double(label.text!)!
            if sender.tag == 12 {//Divide
                label.text = "/"
            }
            if sender.tag == 13 {//Multiply
                label.text = "x"
            }
            if sender.tag == 14 {//Subtract
                label.text = "-"
            }
            if sender.tag == 15 {//Add
                label.text = "+"
            }
            operation = sender.tag
            performingMath = true
        }
        else if sender.tag == 11 {
            label.text = ""
            previousNumber = 0
            numberOnScreen = 0
            performingMath = false
            operation = 0
        }
        else if sender.tag == 16 {
            numberOnScreen = Double(label.text!)!
            if operation == 12 {//Divide
                label.text = String(previousNumber / numberOnScreen)
            }
            if operation == 13 {//Multiply
                label.text = String(previousNumber * numberOnScreen)
            }
            if operation == 14 {//Subtract
                label.text = String(previousNumber - numberOnScreen)
            }
            if operation == 15 {//Add
                label.text = String(previousNumber + numberOnScreen)
            }
        }
    }
}

